### DISCUSSION TITLE
Postgres - 63 chars limit for index name

### DISCUSSION THREAD

**Discussion by proutek** [prisma/prisma#10089] [Q&A]
Hello,

I'm using prisma 3.3.0.

Is there a way to specify name of underlying index in implicit many to many relation. In postgres the index name is cut (63 chars max) but prisma tries to alter name  back to _unique when running migrate dev which leads to error. I know that I can use shorter name for table name but it will complicate migration for us.

![image](https://user-images.githubusercontent.com/21219132/140154604-485e202f-8048-4b6d-9f50-5647c56eb594.png)

**Comment by austincrim** [comment]
Hi @proutek 👋 ,

We allow renaming of indexes using the [`@@index` attribute](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#index) in your schema file.

Try adding something like in the relevant `model` block:
```prisma
@@index(fields: [A, B], name: "MyIndex")
```

**Reply by proutek** [reply]
Thanks for response.

I know about this option but I'm completely lost how to use it in implicit M:N relation. There is no example in documentation for this case.

There is no model for this table in my prisma scheme and I don't want to transform it to explicit one just because prisma can't handle properly long names.

So which model is relevant?

------

```prisma
model BlankArticleVariant {
  id String @id @default(uuid())

  attributeValues BlankArticleVariantAttributeValue[]
}

model BlankArticleVariantAttributeValue {
  id String @id @default(uuid())

  blankArticleVariants BlankArticleVariant[]
}
```

**Reply by austincrim** [reply]
Ah, I see the issue now, thanks for the clarification. I was able recreate the issue locally with these steps:

1. perform a successful initial migration using the schema you posted above
1. add a new field `foo String` in the `BlankArticleVariant` model
1. run `npx prisma migrate dev`
1. see error in console

I'm going to convert this discussion to a GitHub issue where our engineering teams will pick it up for further investigation. Feel free to monitor that issue for updates.

Edit: [here's the issue](https://github.com/prisma/prisma/issues/10094)

**Reply by proutek** [reply]
Just a note. You don't have to do step 2.

**Comment by janpio** [comment]
Follow up question in issue: https://github.com/prisma/prisma/issues/10094#issuecomment-964454823

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
